src/Test/L0/Worker/TaskManagerL0.cs

```diff
@@ -356,130 +356,37 @@ public async void DownloadsTasks()
-                        {
-                            Enabled = true,
-                            Reference = new Pipelines.TaskStepDefinitionReference()
-                            {
-                                Name = bingTaskName,
-                                Version = bingVersion,
-                                Id = bingGuid
-                            }
-                        }
-                    };
-                    using (var stream = GetZipStream())
-                    {
-                        _taskServer
-                            .Setup(x => x.GetTaskContentZipAsync(
-                                bingGuid,
-                                It.Is<TaskVersion>(y => string.Equals(y.ToString(), bingVersion, StringComparison.Ordinal)),
-                                It.IsAny<CancellationToken>()))
-                            .Returns(Task.FromResult<Stream>(stream));
-
-                        //Act
-                        //first invocation will download and unzip the task from mocked IJobServer
-                        await _taskManager.DownloadAsync(_ec.Object, tasks);
-                        //second and third invocations should find the task in the cache and do nothing
-                        await _taskManager.DownloadAsync(_ec.Object, tasks);
-                        await _taskManager.DownloadAsync(_ec.Object, tasks);
+            PreservesTaskZipTask(true, false);
+        }
 
-                        //Assert
-                        //see if the task.json was downloaded
-                        string destDirectory = Path.Combine(
-                            _hc.GetDirectory(WellKnownDirectory.Tasks),
-                            $"{bingTaskName}_{bingGuid}",
-                            bingVersion);
-                        string zipDestDirectory = Path.Combine(_hc.GetDirectory(WellKnownDirectory.TaskZips), $"{bingTaskName}_{bingGuid}_{bingVersion}.zip");
-                        // task.json should exist since we need it for JobExtension.InitializeJob
-                        Assert.True(File.Exists(Path.Combine(destDirectory, Constants.Path.TaskJsonFile)));
-                        // the zip for the task should exist on disk
-                        Assert.True(File.Exists(zipDestDirectory));
-                        //assert download has happened only once, because disabled, duplicate and cached tasks are not downloaded
-                        _taskServer
-                            .Verify(x => x.GetTaskContentZipAsync(It.IsAny<Guid>(), It.IsAny<TaskVersion>(), It.IsAny<CancellationToken>()), Times.Once());
-                    }
-                }
-            }
-            finally
-            {
-                Teardown();
-            }
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void PreservesTaskZipTaskWhenAlwaysExtractTask()
+        {
+            PreservesTaskZipTask(true, false);
```

Comment:
This and the test above are the same thing. We need them to set the Agent config differently, right?

Reply:
I am guessing this should be switched to `false, true`. This would be helped by adding the named parameters.

Reply:
The true false's are different. I like them in 2 separate tests cause if 1 fails is super obvious why. Check out the names. It's just a line in the test so not expensive to do.

Reply:
hahaha yes i'll make sure that's correct @stephenmichaelf !

Reply:
Ah ok it's fixed now! I may not have been looking at latest.

Reply:
Just pushed up the changes!  It was a race condition :-)